tools/ocean_neural_network/canyonb.xml

```diff
@@ -0,0 +1,91 @@
+<tool id="bgc_canyon_b" name="Canyon-B" version="@VERSION@">
+    <description>Robust Estimation of Open Ocean CO2 Variables and Nutrient Concentrations From T, S, and O2 Data Using Bayesian Neural Networks</description>
+    <macros>
+        <token name="@VERSION@">0.9.3</token>
+    </macros>
+    <requirements>
+        <requirement type="package" version="4.3.2">r-base</requirement>
+        <requirement type="package" version="@VERSION@">r-canyonb</requirement>
+    </requirements>
+    <required_files>
+        <include type="literal" path="canyonb.r"/>
+    </required_files>
+    <command detect_errors="exit_code"><![CDATA[ 
+        Rscript
+            '$__tool_directory__/canyonb.r'             
+            '$date'
+            '$latitude'
+            '$longitude'
+            '$pressure'
+            '$temperature'
+            '$salinity'
+            '$dissoxygen'
+            '$variables'
+            '$epressure'
+            '$etemperature'
+            '$epsalinity'
+            '$edissoxygen'
+            '$output'
+        ]]>
+    </command>
+    <inputs>
+        <param name="date" type="text" value="yyyy-mm-dd HH:MM" label="date (UTC) as string ('yyyy-mm-dd HH:MM')"/>
+        <param name="latitude" type="float" min="-90" max="90" value="0"  label="latitude / °N  [-90 90]"/>
+        <param name="longitude" type="float" min="-180" max="180" value="0"  label="longitude / °E [-180 180]"/>
+        <param name="pressure" type="float" value="0"  label="pressure / dbar"/>
+        <param name="temperature" type="float" value="0"  label="in-situ temperature / °C"/>
+        <param name="salinity" type="float" value="0"  label="salinity"/>
+        <param name="dissoxygen" type="float" value="0" label="dissolved oxygen / umol kg-1"/>
+        <param name="variables" type="select" label="Variable names to be predicted" multiple="true">
+                <option value="AT">AT</option>
+                <option value="CT">CT</option>
+                <option value="pH">pH</option>
+                <option value="pCO2">pCO2</option>
+                <option value="NO3">NO3</option>
+                <option value="PO4">PO4</option>
+                <option value="SiOH4">SiOH4</option>
+        </param>
+        <param name="epressure" type="float" value="0.5" label="Default value for input errors pressure (optional) 0.5 dbar" optional="true"/>
```

Comment:
What is dbar. Can we add max min?

Reply:
If I remember my physic classes dbar (decibar) = 0,1 bar the pressure unit ;) I am pretty sure we can put min and max as it (same if I remember my classes well) is the acceptable error. But I'll have to ask my coworker the exact min and max we can put